darwin/Classes/UniversalBlePlugin.swift

```diff
@@ -262,8 +262,19 @@ private class BleCentralDarwin: NSObject, UniversalBlePlatformChannel, CBCentral
     completion(Result.failure(FlutterError(code: "NotSupported", message: nil, details: nil)))
   }
 
-  func pair(deviceId _: String) throws {
-    throw FlutterError(code: "NotSupported", message: nil, details: nil)
+  func pair(deviceId: String) throws {
+    guard let peripheral = discoveredPeripherals[deviceId] else { return }
+    
+    guard peripheral.state == .connected else { throw FlutterError(code: "NotConnected", message: nil, details: nil)} // TODO: Connect before attempting to pair, to align the behavior with the rest of the platforms
+    guard let services = peripheral.services else { throw FlutterError(code: "ServicesNotDiscovered", message: nil, details: nil)}
+
+    services.forEach { service in
+      service.characteristics?.forEach { characteristic in
+        if characteristic.properties.contains(.read) {
+          peripheral.readValue(for: characteristic)
+        }
```

Comment:
@fotiDim i think there are many things to handle natively
1: connection will get update in callback, so its tricky to connect natively and perform rest of the operation
2: services results will again gets in two callbacks, one for service discovery and second for their characteristic discovery, so this is even more tricky

another option is we keep it as it is, and expects that plugin users will auto handle the Connection and Discovery first
and we are just trying to perform a read requset, 

Now this creates some additional handleing to perform this operation on Apple platforms, and hard to manage cross platforms
because rest of the paltforms can Pair without connection ( or even if they connect first, they do it internally )

also if a device does not have any encrypted char, this will fail, and will result into wrong info for the Api

So i think it would be better to just inform in Docs, that they can try to read/write to an encrypted characteristic on Apple platforms to trigger pair process

Reply:
The ToDo is for the future. I didn't mean to tackle it in this PR. I know it is hard, but it is not impossible.

The pair() API does not return any value back, so we won't report any inaccurate info. It's just not guaranteed to work on Apple platforms. I suppose the same applies for the rest of the platforms as not all peripherals are expected to respond to pairing requests.

I added documentation and exceptions for the above. If you think that they could be improved let's do it. My opinion is that it is better to have this logic in the library rather than the app. We could add a parameter in the pair() command to pass a service/characteristic if we know already which one we want to use to pair with and avoid iterating over all of them.

Reply:
We can detect if pairing succeeded or failed on supported platforms, all supported platforms provides a callback if paired or failed

but still if we want to implement this on Apple, then i would suggest to implement on dart side instead on native
because on dart side, we already have the required things, we can call for discover services asynchronously
we can easily wait for the connection event on dart side, thanks to the dart's apis to wait for a callback completion

and finally we can call read on all the services
so i think it would be much better on dart side

Reply:
We can't detect pairing on Apple. I gave that a try. isPaired will always return null on Apple.

Reply:
On Apple, I dont think there is a way to detect the pairing result. It will always be unknown.